pghoard/transfer.py

```diff
@@ -39,6 +40,105 @@ class TransferOperation(StrEnum):
     Metadata = "metadata"
 
 
+class TransferIncrement(NamedTuple):
+    n_bytes: float
+    tracked_at: float
+
+
+class OperationEventProgressTracker(PGHoardThread):
```

Comment:
**question** Not sure about the naming yet, since currently I am planning to use it only for upload due to the scope of the task. Maybe `UpdateEventProgressTracker`?